sematic/api_client.py

```diff
@@ -986,3 +987,85 @@ def _get_encoded_query_filters(filters: Dict[str, Any]) -> Optional[str]:
         return json.dumps({"AND": [{col: {"eq": filters[col]}} for col in filters]})
 
     return None
+
+
+def block_on_run(
+    run_id: str, delay_seconds: float = 5.0, max_wait_seconds: Optional[float] = None
+) -> None:
+    """Block on the run with the given id until it is in a terminal state.
+
+    Terminal states include successful completion, failure and cancellation.
+    Only successful completion returns without error, other terminal states
+    result in `RuntimeError` being raised.
+
+    Parameters
+    ----------
+    run_id:
+        The id of the run to block on.
+    delay_seconds:
+        The number of seconds between polling for updates to the run's
+        status.
+    max_wait_seconds:
+        If the run has not terminated after this number of seconds, will raise
+        `TimeoutError`. If this is `None`, will wait indefinitely. Note that
+        if `block_on_run` has failed with a timeout, this does NOT mean the run
+        itself has failed or timed out; the run may continue unimpacted. This
+        timeout is only for the block call itself.
+    """
+    terminal_state = None
+    start_time = time.time()
+    while terminal_state is None:
+        state = get_run(run_id).future_state
+
+        if isinstance(state, str):
+            state = FutureState[state]
+
+        if state.is_terminal():
+            terminal_state = state
+        else:
+            time.sleep(delay_seconds)
+        if max_wait_seconds is not None and time.time() - start_time > max_wait_seconds:
+            raise TimeoutError(
+                f"Run {run_id} did not complete in {max_wait_seconds} seconds"
+            )
+
+    if terminal_state is not FutureState.RESOLVED:
+        raise RuntimeError("Run did not complete successfully")
```

Comment:
Please also include the actual state value in the message.

Reply:
done